haystack/tools/parameters_schema_utils.py

```diff
@@ -187,7 +188,7 @@ def _resolve_type(_type: Any) -> Any:  # noqa: PLR0911  # pylint: disable=too-ma
     if origin is collections.abc.Sequence:
         return Sequence[_resolve_type(args[0]) if args else Any]  # type: ignore[misc]
 
-    if origin is Union:
+    if is_union_type(origin):
         return Union[tuple(_resolve_type(a) for a in args)]
```

Comment:
This is probably fine as long as Pydantic continues to support the old types like `Union`. But we may need to keep an eye on this if Pydantic ever drops support and switch to constructing this with `|`